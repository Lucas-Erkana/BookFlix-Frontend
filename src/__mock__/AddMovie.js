import React, { useState } from 'react';
import { BrowserRouter, useNavigate } from 'react-router-dom';
import { storage } from '../firebaseConfig';

const AddMovie = () => {
  const [name, setName] = useState('');
  const [price, setPrice] = useState('');
  const [image, setImage] = useState('');
  const [details, setDetails] = useState('');
  const [uploadProgress, setUploadProgress] = useState(0);
  const [isUploading, setIsUploading] = useState(false);
  const [uploadSuccess, setUploadSuccess] = useState(false);
  const [errorMessage, setErrorMessage] = useState('');
  const [duration, setDuration] = useState('');


  const handleFileChange = (e) => {
    const file = e.target.files[0];
    setImage(file);
  };

  const handleAddClick = () => {
    if (!name || !price || !image || !details) {
      setErrorMessage('Please fill in all fields');
      return;
    }

    const allowedExtensions = /\.(jpg|jpeg|png)$/i;
    if (!allowedExtensions.test(image.name)) {
      setErrorMessage('Invalid file format');
      return;
    }

    const storageRef = storage.ref();
    const fileRef = storageRef.child(image.name);

    setIsUploading(true);
    const uploadTask = fileRef.put(image);

    uploadTask.on(
      'state_changed',
      (snapshot) => {
        const progress = Math.round((snapshot.bytesTransferred / snapshot.totalBytes) * 100);
        setUploadProgress(progress);
      },
      (error) => {
        setErrorMessage(error.message);
        setIsUploading(false);
      },
      () => {
        uploadTask.snapshot.ref.getDownloadURL().then((downloadURL) => {
          const newMovie = {
            name,
            price,
            image: downloadURL,
            details,
            duration,
          };

          setName('');
          setPrice('');
          setImage('');
          setDetails('');
          setDuration('');
          setIsUploading(false);
          setUploadSuccess(true);
          navigate('/');
        });
      },
    );
  };
  const handleCancelClick = () => {
    navigate('/');
  };

  return (
    <BrowserRouter>
      <div className="col-md col container-fluid d-flex justify-content-center align-items-center add-movie-container">
        <div className="d-flex flex-column m-5 p-5 align-items-center add-movie-div">
          <div className="d-sm-inline-flex align-items-center my-4">
            <h1 className="mx-3">Movie</h1>
          </div>

          <form className="row g-4 add-movie-form">
            <div className="col-md-12">
              <input type="text" placeholder="Movie Type" className="form-control" value={name} onChange={(e) => setName(e.target.value)} />
            </div>

            <div className="input-group col-md-12">
              <span className="input-group-text add-price-input">$</span>
              <input type="float" placeholder="Movie Charge" className="form-control" value={price} onChange={(e) => setPrice(e.target.value)} />
              <span className="input-group-text add-price-input">.00</span>
            </div>
            <div className="col-md-12">
              <input type="textarea" placeholder="Details" className="form-control" value={details} onChange={(e) => setDetails(e.target.value)} />
            </div>
            <div className="col-md-12">
              <input type="file" className="form-control" id="inputGroupFile02" onChange={handleFileChange} />
            </div>
            <div className="col-md-4 d-flex">
              <input type="number" placeholder="Duration" className="form-control" value={duration} onChange={(e) => setDuration(e.target.value)} />
            </div>
            <div className="col-md-12 d-flex justify-content-end g-3">
              <button type="button" className="col-md-5 btn btn-success add-btn mx-3" onClick={handleAddClick}>Add Movie</button>
              <button type="button" className="col-md-4 btn btn-outline-danger cancel-add-btn" onClick={handleCancelClick}>Cancel</button>
            </div>

            {isUploading && (
              <div>
                <p>
                  Uploading:
                  {uploadProgress}
                  %
                </p>
              </div>
            )}

            {uploadSuccess && (
              <div>
                <p>File uploaded successfully!</p>
              </div>
            )}

            {errorMessage && (
              <div>
                <p>{errorMessage}</p>
              </div>
            )}
          </form>
        </div>
      </div>
    </BrowserRouter>
  );
};

export default AddMovie;
