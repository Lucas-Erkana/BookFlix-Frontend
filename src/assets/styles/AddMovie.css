.add-movie-container {
  background:
    linear-gradient(0deg, rgb(146 185 10 / 88%), rgb(146 185 12 / 88%)),
    url("../images/add-movie.jpg");
  background-repeat: no-repeat;
  background-size: cover;
  height: 100vh;
  background-position: center;
}

.add-movie-div {
  width: 70%;
  border-radius: 1rem;
}

.add-movie-form .form-control {
  border-radius: 2rem;
}

.add-btn {
  background-color: #98bf11;
  color: white;
  border-radius: 20px;
}

.cancel-add-btn {
  border-radius: 20px;
}
