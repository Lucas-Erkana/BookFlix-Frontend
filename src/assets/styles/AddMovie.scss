.add-movie-container {
  background:
    linear-gradient(0deg, rgb(146 185 10 / 88%), rgb(146 185 12 / 88%)),
    url("../images/reservation.png");
  background-repeat: no-repeat;
  background-size: cover;
  height: 100vh;
  background-position: center;
}

.add-movie-div {
  width: 70%;
  border-radius: 1rem;
}

.form-control {
  background: transparent;
  border: 2px solid white;
}

.form-control::placeholder {
  color: white;
}

.form-control:focus {
  border: white;
  box-shadow: none;
}

.form-control:focus::placeholder {
  color: #343a40 !important;
}

.add-movie-form .form-control {
  border-radius: 2rem;
}

.add-movie,
.cancel-add {
  padding: 0.375rem 3rem;
}

.add-movie {
  background-color: transparent;
  border: 1px solid white;
  color: white;
}

.add-movie:hover {
  background-color: white;
  color: rgb(25, 135, 84);
}

.cancel-add {
  background-color: transparent;
  border: 1px solid white;
  color: white;
}

.cancel-add:hover {
  background-color: #dc3545;
  color: white;
}

.add-price-input-l, .add-price-input-r{
  background-color: transparent;
  border-radius: 2rem;
  border: 2px solid white;
}

.add-price-input-l {
  border-right: none;
}

.add-price-input-r {
  border-left: none;
}

.ticket-price{
  border-left: none;
  border-right: none;
}
