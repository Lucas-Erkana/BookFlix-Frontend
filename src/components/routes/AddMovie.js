import React, { useState } from 'react';
import { useDispatch } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import { addMovie } from '../../redux/MoviesSlice';
import { storage } from '../../firebaseConfig';

const AddMovie = () => {
  const [name, setName] = useState('');
  const [price, setPrice] = useState('');
  const [image, setImage] = useState('');
  const [details, setDetails] = useState('');
  const [trailer, setTrailer] = useState('');
  const [uploadProgress, setUploadProgress] = useState(0);
  const [isUploading, setIsUploading] = useState(false);
  const [uploadSuccess, setUploadSuccess] = useState(false);
  const [errorMessage, setErrorMessage] = useState('');
  const [duration, setDuration] = useState('');

  const dispatch = useDispatch();
  const navigate = useNavigate();

  const handleFileChange = (e) => {
    const file = e.target.files[0];
    setImage(file);
  };

  const handleAddClick = () => {
    if (!name || !price || !image || !details || !trailer) {
      setErrorMessage('Please fill in all fields');
      return;
    }

    const allowedExtensions = /\.(jpg|jpeg|png)$/i;
    if (!allowedExtensions.test(image.name)) {
      setErrorMessage('Invalid file format');
      return;
    }

    const storageRef = storage.ref();
    const fileRef = storageRef.child(image.name);

    setIsUploading(true);
    const uploadTask = fileRef.put(image);

    uploadTask.on(
      'state_changed',
      (snapshot) => {
        const progress = Math.round((snapshot.bytesTransferred / snapshot.totalBytes) * 100);
        setUploadProgress(progress);
      },
      (error) => {
        setErrorMessage(error.message);
        setIsUploading(false);
      },
      () => {
        uploadTask.snapshot.ref.getDownloadURL().then((downloadURL) => {
          const newMovie = {
            name,
            price,
            image: downloadURL,
            details,
            trailer,
            duration,
          };
          dispatch(addMovie(newMovie));
          setName('');
          setPrice('');
          setImage('');
          setDetails('');
          setTrailer('');
          setDuration('');
          setIsUploading(false);
          setUploadSuccess(true);
          navigate('/');
        });
      },
    );
  };
  const handleCancelClick = () => {
    navigate('/');
  };

  return (
    <div className="col-md col container-fluid d-flex justify-content-center align-items-center add-movie-container">
      <div className="d-flex flex-column m-5 align-items-center add-movie-div">
        <h2 className="text-white w-100 text-center my-3">Add New Movie</h2>

        <form className="row g-4 add-movie-form">
          <div className="col-md-12">
            <input type="text" placeholder="Movie title" className="form-control" value={name} onChange={(e) => setName(e.target.value)} />
          </div>

          <div className="input-group col-md-12">
            <span className="input-group-text add-price-input-l">$</span>
            <input type="float" placeholder="Ticket price" className="form-control ticket-price" value={price} onChange={(e) => setPrice(e.target.value)} />
            <span className="input-group-text add-price-input-r">.00</span>
          </div>
          <div className="col-md-12">
            <input type="textarea" placeholder="Movie Details" className="form-control" value={details} onChange={(e) => setDetails(e.target.value)} />
          </div>
          <div className="col-md-12 d-flex flex-md-row flex-column justify-content-between gap-4">
            <input type="file" className="form-control" id="inputGroupFile02" onChange={handleFileChange} />
            <input type="number" placeholder="Duration" className="form-control" value={duration} onChange={(e) => setDuration(e.target.value)} />
          </div>
          <div className="col-md-12">
            <input type="textarea" placeholder="trailer" className="form-control" value={trailer} onChange={(e) => setTrailer(e.target.value)} />
          </div>
          <div className="col-md-12 d-flex justify-content-center">
            <button type="button" className="col-md-3 btn rounded-pill mx-3 add-movie" onClick={handleAddClick}>Add Movie</button>
            <button type="button" className="col-md-3 btn rounded-pill cancel-add" onClick={handleCancelClick}>Cancel</button>
          </div>

          {isUploading && (
          <div>
            <p className="text-light text-center text-warning">
              Uploading:
              {uploadProgress}
              %
            </p>
          </div>
          )}

          {uploadSuccess && (
          <div className="text-center text-success">
            <p>File uploaded successfully!</p>
          </div>
          )}

          {errorMessage && (
          <div className="text-center text-warning">
            <p>{errorMessage}</p>
          </div>
          )}
        </form>
      </div>
    </div>
  );
};

export default AddMovie;
