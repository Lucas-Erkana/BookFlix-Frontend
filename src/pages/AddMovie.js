import React, { useState } from 'react';
import { useDispatch } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import { addMovie } from '../store/MoviesSlice';
import { storage } from '../firebaseConfig';

const AddMovie = () => {
  const [name, setName] = useState('');
  const [price, setPrice] = useState('');
  const [image, setImage] = useState('');
  const [details, setDetails] = useState('');
  const [trailer, setTrailer] = useState('');
  const [uploadProgress, setUploadProgress] = useState(0);
  const [isUploading, setIsUploading] = useState(false);
  const [uploadSuccess, setUploadSuccess] = useState(false);
  const [errorMessage, setErrorMessage] = useState('');
  const [duration, setDuration] = useState('');

  const dispatch = useDispatch();
  const navigate = useNavigate();

  const handleFileChange = (e) => {
    const file = e.target.files[0];
    setImage(file);
  };

  const handleAddClick = () => {
    if (!name || !price || !image || !details || !trailer) {
      setErrorMessage('Please fill in all fields');
      return;
    }

    const allowedExtensions = /\.(jpg|jpeg|png)$/i;
    if (!allowedExtensions.test(image.name)) {
      setErrorMessage('Invalid file format');
      return;
    }

    const storageRef = storage.ref();
    const fileRef = storageRef.child(image.name);

    setIsUploading(true);
    const uploadTask = fileRef.put(image);

    uploadTask.on(
      'state_changed',
      (snapshot) => {
        const progress = Math.round((snapshot.bytesTransferred / snapshot.totalBytes) * 100);
        setUploadProgress(progress);
      },
      (error) => {
        setErrorMessage(error.message);
        setIsUploading(false);
      },
      () => {
        uploadTask.snapshot.ref.getDownloadURL().then((downloadURL) => {
          const newMovie = {
            name,
            price,
            image: downloadURL,
            details,
            trailer,
            duration,
          };
          dispatch(addMovie(newMovie));
          setName('');
          setPrice('');
          setImage('');
          setDetails('');
          setTrailer('');
          setDuration('');
          setIsUploading(false);
          setUploadSuccess(true);
          navigate('/');
        });
      },
    );
  };
  const handleCancelClick = () => {
    navigate('/');
  };

  return (
    <div className="col-md col container-fluid d-flex justify-content-center align-items-center add-movie-container">
      <div className="d-flex flex-column m-5 align-items-center add-movie-div">
        <h2 className="text-white w-100 text-center my-3">Add New Movie</h2>

        <form className="row g-4 add-movie-form">
          <div className="col-md-12">
            <input type="text" placeholder="Movie Type" className="form-control" value={name} onChange={(e) => setName(e.target.value)} />
          </div>

          <div className="input-group col-md-12">
            <span className="input-group-text add-price-input">$</span>
            <input type="float" placeholder="Movie Charge" className="form-control" value={price} onChange={(e) => setPrice(e.target.value)} />
            <span className="input-group-text add-price-input">.00</span>
          </div>
          <div className="col-md-12">
            <input type="textarea" placeholder="Details" className="form-control" value={details} onChange={(e) => setDetails(e.target.value)} />
          </div>
          <div className="col-md-12">
            <input type="textarea" placeholder="trailer" className="form-control" value={trailer} onChange={(e) => setTrailer(e.target.value)} />
          </div>
          <div className="col-md-12 d-flex flex-md-row flex-column justify-content-between g-4">
            <input type="file" className="form-control" id="inputGroupFile02" onChange={handleFileChange} />
            <input type="number" placeholder="Duration" className="form-control" value={duration} onChange={(e) => setDuration(e.target.value)} />
          </div>
          <div className="col-md-12 d-flex justify-content-end g-3">
            <button type="button" className="col-md-5 btn btn-light text-success rounded-pill mx-3" onClick={handleAddClick}>Add Movie</button>
            <button type="button" className="col-md-4 btn btn-danger rounded-pill" onClick={handleCancelClick}>Cancel</button>
          </div>

          {isUploading && (
          <div>
            <p className="text-light">
              Uploading:
              {uploadProgress}
              %
            </p>
          </div>
          )}

          {uploadSuccess && (
          <div>
            <p>File uploaded successfully!</p>
          </div>
          )}

          {errorMessage && (
          <div>
            <p>{errorMessage}</p>
          </div>
          )}
        </form>
      </div>
    </div>
  );
};

export default AddMovie;
