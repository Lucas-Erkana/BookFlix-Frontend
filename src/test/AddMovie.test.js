import { render, cleanup } from '@testing-library/react';
import AddMovie from '../__mock__/AddMovie';

afterEach(() => {
  cleanup();
});

describe('AddMovie', () => {
  it('AddMovie renders correctly', () => {
    const addmovie = render(<AddMovie />);
    expect(addmovie).toMatchSnapshot();
  });
});
