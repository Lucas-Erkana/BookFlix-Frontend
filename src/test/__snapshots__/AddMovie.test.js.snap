// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddMovie AddMovie renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="col-md col container-fluid d-flex justify-content-center align-items-center add-service-container"
      >
        <div
          class="d-flex flex-column m-5 p-5 align-items-center add-service-div"
        >
          <div
            class="d-sm-inline-flex align-items-center my-4"
          >
            <h1
              class="mx-3"
            >
              Service
            </h1>
          </div>
          <form
            class="row g-4 add-service-form"
          >
            <div
              class="col-md-12"
            >
              <input
                class="form-control"
                placeholder="Service Type"
                type="text"
                value=""
              />
            </div>
            <div
              class="input-group col-md-12"
            >
              <span
                class="input-group-text add-price-input"
              >
                $
              </span>
              <input
                class="form-control"
                placeholder="Service Charge"
                type="float"
                value=""
              />
              <span
                class="input-group-text add-price-input"
              >
                .00
              </span>
            </div>
            <div
              class="col-md-12"
            >
              <input
                class="form-control"
                placeholder="Details"
                type="textarea"
                value=""
              />
            </div>
            <div
              class="col-md-12"
            >
              <input
                class="form-control"
                id="inputGroupFile02"
                type="file"
              />
            </div>
            <div
              class="col-md-4 d-flex"
            >
              <input
                class="form-control"
                placeholder="Duration"
                type="number"
                value=""
              />
            </div>
            <div
              class="col-md-12 d-flex justify-content-end g-3"
            >
              <button
                class="col-md-5 btn btn-success add-btn mx-3"
                type="button"
              >
                Add Service
              </button>
              <button
                class="col-md-4 btn btn-outline-danger cancel-add-btn"
                type="button"
              >
                Cancel
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="col-md col container-fluid d-flex justify-content-center align-items-center add-service-container"
    >
      <div
        class="d-flex flex-column m-5 p-5 align-items-center add-service-div"
      >
        <div
          class="d-sm-inline-flex align-items-center my-4"
        >
          <h1
            class="mx-3"
          >
            Service
          </h1>
        </div>
        <form
          class="row g-4 add-service-form"
        >
          <div
            class="col-md-12"
          >
            <input
              class="form-control"
              placeholder="Service Type"
              type="text"
              value=""
            />
          </div>
          <div
            class="input-group col-md-12"
          >
            <span
              class="input-group-text add-price-input"
            >
              $
            </span>
            <input
              class="form-control"
              placeholder="Service Charge"
              type="float"
              value=""
            />
            <span
              class="input-group-text add-price-input"
            >
              .00
            </span>
          </div>
          <div
            class="col-md-12"
          >
            <input
              class="form-control"
              placeholder="Details"
              type="textarea"
              value=""
            />
          </div>
          <div
            class="col-md-12"
          >
            <input
              class="form-control"
              id="inputGroupFile02"
              type="file"
            />
          </div>
          <div
            class="col-md-4 d-flex"
          >
            <input
              class="form-control"
              placeholder="Duration"
              type="number"
              value=""
            />
          </div>
          <div
            class="col-md-12 d-flex justify-content-end g-3"
          >
            <button
              class="col-md-5 btn btn-success add-btn mx-3"
              type="button"
            >
              Add Service
            </button>
            <button
              class="col-md-4 btn btn-outline-danger cancel-add-btn"
              type="button"
            >
              Cancel
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
